### PR TITLE
Adds how-to guide on adding authorities to access tokens

### PR DESCRIPTION
**Pull request by ddubson**
Closes issue #542

### CODE REVIEW

File: docs/src/docs/asciidoc/guides/how-to-custom-claims-authorities.adoc [hunk 1]
```diff
@@ -0,0 +1,49 @@
+[[how-to-extension-grant-type]]
+= How-to: Add authorities to custom claims in JWT-based access tokens
```
Comment (jgrandja) [inline docs/src/docs/asciidoc/guides/how-to-custom-claims-authorities.adoc:2]: Please update to -> "How-to: Add authorities as custom claims in JWT access tokens"

File: docs/src/docs/asciidoc/guides/how-to-custom-claims-authorities.adoc [hunk 2]
```diff
@@ -0,0 +1,49 @@
+[[how-to-extension-grant-type]]
```
Comment (jgrandja) [inline docs/src/docs/asciidoc/guides/how-to-custom-claims-authorities.adoc:1]: Update to `how-to-custom-claims-authorities`

File: docs/src/docs/asciidoc/guides/how-to-custom-claims-authorities.adoc [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+= How-to: Add authorities to custom claims in JWT-based access tokens
+:index-link: ../how-to.html
+:docs-dir: ..
+:examples-dir: {docs-dir}/examples
+
+This guide demonstrates how to append resource owner authorities to a JSON Web Token (JWT)-based access token.
```
Comment (jgrandja) [inline docs/src/docs/asciidoc/guides/how-to-custom-claims-authorities.adoc:7]: append -> add
JSON Web Token (JWT)-based access token -> JWT access token

File: docs/src/docs/asciidoc/guides/how-to-custom-claims-authorities.adoc [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+The term 'authorities' may be described in varying forms such as roles, permissions, or groups of the resource owner.
```
Comment (jgrandja) [inline docs/src/docs/asciidoc/guides/how-to-custom-claims-authorities.adoc:8]: Use double quotes for 'authorities'

may be described in varying forms -> may represent varying forms

File: docs/src/docs/asciidoc/guides/how-to-custom-claims-authorities.adoc [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+
+To make resource owners' authorities available to the resource server, we add custom claims to an access token issued by Spring Authorization Server.
```
Comment (jgrandja) [inline docs/src/docs/asciidoc/guides/how-to-custom-claims-authorities.adoc:10]: resource owners' -> resource owner's

File: docs/src/docs/asciidoc/examples/src/main/java/sample/customClaims/authorities/CustomClaimsWithAuthoritiesConfiguration.java [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+				.build();
+		return new InMemoryUserDetailsManager(user);
+	}
+
+	@Bean
+	public OAuth2TokenCustomizer<JwtEncodingContext> jwtTokenCustomizer() { // <3>
+		return (context) -> {
+			if (OAuth2TokenType.ACCESS_TOKEN.equals(context.getTokenType())) { // <4>
+				context.getClaims().claims((claims) -> { // <5>
+					var roles = AuthorityUtils.authorityListToSet(context.getPrincipal().getAuthorities())
```
Comment (jgrandja) [inline docs/src/docs/asciidoc/examples/src/main/java/sample/customClaims/authorities/CustomClaimsWithAuthoritiesConfiguration.java:34]: Replace `var` with `Set<String>`

File: docs/src/docs/asciidoc/guides/how-to-custom-claims-authorities.adoc [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+
+We define a sample user with a mix of authorities for demonstration purposes, and populate custom claims in an access token
+with those authorities.
+
+[[sample.customClaims.authorities]]
+include::code:CustomClaimsWithAuthoritiesConfiguration[]
+
+<1> Define a sample user `user1` within an internal user details service.
+<2> Define a few roles for  `user1`.
+<3> Define `OAuth2TokenCustomizer<JwtEncodingContext>` bean that allows for customizing JWT token claims.
```
Comment (jgrandja) [inline docs/src/docs/asciidoc/guides/how-to-custom-claims-authorities.adoc:42]: bean -> `@Bean`

File: docs/src/docs/asciidoc/guides/how-to-custom-claims-authorities.adoc [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+<4> Check whether the JWT token is an access token.
+<5> From the encoding context, modify the claims of the access token.
+<6> Extract user roles from the `Principal` object. The role information for internal users is stored as a string prefixed with `ROLE_`, so we strip the prefix here.
+<7> Set custom claim `roles` to the set of roles collected from the previous step.
+
+As a result of this customization, authorities information about the user will be included as a custom claim within an
```
Comment (jgrandja) [inline docs/src/docs/asciidoc/guides/how-to-custom-claims-authorities.adoc:48]: within an access token -> within the access token

**Review by jgrandja — CHANGES_REQUESTED** [review]
Thanks for the PR @ddubson ! Please see review comments.

Also, please rebase on `main`. Thanks!

**Comment by ddubson** [comment]
@jgrandja Thanks for the review! I've addressed the points outlined and rebased onto the latest Antora-based docs structure. Antora rendering looks great!

**Comment by jgrandja** [comment]
Thanks for the updates @ddubson ! This is now merged along with a polish commit.

**Pull request closed without merging**